src/domains/domu/domu_cfg.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2024 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <domain.h>
+#include <domains/dom_runner.h>
+#include <string.h>
+#include <xen_dom_mgmt.h>
+#include <xl_parser.h>
+
+#include <zephyr/xen/public/domctl.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(domains);
+
+extern char __img_domu_start[];
+extern char __img_domu_end[];
+extern char __dtb_domu_start[];
+extern char __dtb_domu_end[];
+
+static int load_domu_image(uint8_t* buf, size_t bufsize, uint64_t image_load_offset, void* image_info)
+{
+    ARG_UNUSED(image_info);
+    memcpy(buf, __img_domu_start + image_load_offset, bufsize);
+    return 0;
+}
+
+static ssize_t get_domu_image_size(void* image_info, uint64_t* size)
+{
+    ARG_UNUSED(image_info);
+    *size = __img_domu_end - __img_domu_start;
+    return 0;
+}
+
+static const char *domu_backend_configs[] = {
+    "disk=['backend=1, vdev=xvda, access=rw, target=/dev/sda7']",
+    "vif=['backend=1,bridge=xenbr0,mac=08:00:27:ff:cb:ce,ip=172.44.0.2 255.255.255.0 172.44.0.1']",
+};
+
+struct xen_domain_cfg domu_cfg = {
+    .mem_kb            = 0x40000, /* 256Mb */
+
+    .flags               = (XEN_DOMCTL_CDF_hvm | XEN_DOMCTL_CDF_hap | XEN_DOMCTL_CDF_iommu),
```

Comment:
XEN_DOMCTL_CDF_iommu - Are you sure it's needed for DomU? RPI5 for example will just fail with this flag set.

Reply:
I believe it is not needed in this particular case, because this DomU has no dtdevs assigned for it.  On other hand, having this option will not break anything